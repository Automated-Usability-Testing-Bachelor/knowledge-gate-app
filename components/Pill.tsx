import {StyleSheet, Text, View} from "react-native";

function Pill({backgroundColor, borderColor, fontColor, text}: any) {
    return (
        <View style={[
            styles.container,
            {
                backgroundColor: backgroundColor,
                borderColor: borderColor,
                borderWidth: 1,
            }]}>
            <Text style={[styles.text, {color: fontColor}]}>{text}</Text>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        borderRadius: 100,
        // justifyContent: "center",
        alignItems: "center",
        // height: "auto",
        // width: "auto",
        // alignSelf: "flex-start",
        paddingVertical: 5,
        paddingHorizontal: 10,
        margin: 5,
    },
    text: {
        fontSize: 12,
        fontFamily: "Sans-Regular",
    },
});

export default Pill;