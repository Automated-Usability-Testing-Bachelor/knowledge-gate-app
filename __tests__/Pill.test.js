/*eslint-disable */
import React from 'react'
import renderer from 'react-test-renderer'
import Pill from '../components/Pill'

// eslint-disable-next-line no-undef
describe('<Pill />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<Pill />).toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot()
  });
});

