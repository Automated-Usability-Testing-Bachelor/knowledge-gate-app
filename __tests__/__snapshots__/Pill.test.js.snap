// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> has 1 child 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 100,
        "margin": 2.5,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      },
      {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderWidth": 1,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Sans-Regular",
          "fontSize": 12,
        },
        {
          "color": undefined,
        },
      ]
    }
  />
</View>
`;
